File: 0792-binary-search/0792-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l , r = 0 , len(nums)-1
        while l <= r:
            mid = (l + r) // 2
            #mid = (r - l) // 2 + l 不會出現溢位
            if target == nums[mid]:
                return mid
            elif target > nums[mid]:
                l = mid + 1
            else:
                r = mid - 1
        return -1

File: 0792-binary-search/0792-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0 , r = nums.length - 1 ;
        while(l <= r){
            int mid = (r -l) / 2 + l;
            if(target == nums[mid]){
                return mid;
            }else if(target > nums[mid]){
                l = mid + 1;
            }else{
                r = mid - 1;
            }
        }
        return -1;
    }
}